crates/csharp/src/interface.rs

```diff
@@ -168,14 +196,211 @@ impl InterfaceGenerator<'_> {
+                [global::System.Runtime.InteropServices.DllImportAttribute("{import_module_name}", EntryPoint = "[future-drop-writeable-0][async]{future}"), global::System.Runtime.InteropServices.WasmImportLinkageAttribute]
+                internal static extern void {camel_name}DropWriteable(int writeable);
+                "#
+            );
+
+            self.csharp_gen
+                .interface_fragments
+                .entry(self.name.to_string())
+                .or_insert_with(|| InterfaceTypeAndFragments::new(false))
+                .interface_fragments
+                .push(InterfaceFragment {
+                    csharp_src: format!(r#"
+                    public static (FutureReader, FutureWriter) {camel_name}VoidNew() 
+                    {{
+                         var packed = {interop_name}.{camel_name}VoidNew();
+                         var readerHandle = (int)(packed & 0xFFFFFFFF);
+                         var writerHandle = (int)(packed >> 32);
+
+                         return (new FutureReaderVoid(readerHandle), new FutureWriterVoid(writerHandle));
+                    }}
+                    "#).to_string(),
+                    csharp_interop_src: "".to_string(),
+                    stub: "".to_string(),
+                    direction: Some(self.direction),
+                });
+
+            self.csharp_gen.needs_future_reader_support = true;
+            self.csharp_gen.needs_future_writer_support = true;
+        }
+
+        uwrite!(
+            self.csharp_interop_src,
+            r#"
+                [global::System.Runtime.InteropServices.DllImportAttribute("$root", EntryPoint = "[waitable-set-new]"), global::System.Runtime.InteropServices.WasmImportLinkageAttribute]
+                internal static extern int WaitableSetNew();
+                "#
+        );
+
+        uwrite!(
+            self.csharp_interop_src,
+            r#"
+                [global::System.Runtime.InteropServices.DllImportAttribute("$root", EntryPoint = "[waitable-join]"), global::System.Runtime.InteropServices.WasmImportLinkageAttribute]
+                internal static extern void WaitableJoin(int waitable, int set);
+                "#
+        );
+
+        uwrite!(
+            self.csharp_interop_src,
+            r#"
+                [global::System.Runtime.InteropServices.DllImportAttribute("$root", EntryPoint = "[waitable-set-wait]"), global::System.Runtime.InteropServices.WasmImportLinkageAttribute]
+                internal static unsafe extern int WaitableSetWait(int waitable, int* waitableHandlePtr);
+                "#
+        );
+
+        uwrite!(
+            self.csharp_interop_src,
+            r#"
+                [global::System.Runtime.InteropServices.DllImportAttribute("$root", EntryPoint = "[waitable-set-drop]"), global::System.Runtime.InteropServices.WasmImportLinkageAttribute]
+                internal static unsafe extern void WaitableSetDrop(int waitable);
+                "#
+        );
+
```

Comment:
These could be moved to a standalone `.cs` file that gets copied into the output since they don't depend on the WIT in any way.

Reply:
Done.[](url)